### PR TITLE
Clear flow event after cancel

### PR DESCRIPTION
**Pull request by umustafi**
Dear Gobblin maintainers,

Please accept this PR. I understand that it will not be reviewed until I have checked off all the steps below!

### JIRA
- [ ] My PR addresses the following [Gobblin JIRA](https://issues.apache.org/jira/browse/GOBBLIN/) issues and references them in the PR title. For example, "[GOBBLIN-XXX] My Gobblin PR"
    - https://issues.apache.org/jira/browse/GOBBLIN-1436

### Description
- [ ] Here are some details about my PR, including screenshots (if applicable):
After the job emits cancel event it needs to have the Flow Event field cleared, so it can have its job status marked as Resumed later. 

### Tests
- [X] My PR adds the following unit tests __OR__ does not need testing for this extremely good reason:

### Commits
- [ ] My commits all reference JIRA issues in their subject lines, and I have squashed multiple commits if they address the same issue. In addition, my commits follow the guidelines from "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)":
    1. Subject is separated from body by a blank line
    2. Subject is limited to 50 characters
    3. Subject does not end with a period
    4. Subject uses the imperative mood ("add", not "adding")
    5. Body wraps at 72 characters
    6. Body explains "what" and "why", not "how"

### CODE REVIEW

**Review by Will-Lo — APPROVED** [review]

**Comment by Will-Lo** [comment]
@jack-moseley could you take a look?

**Review by autumnust — APPROVED** [review]
+1

File: gobblin-service/src/main/java/org/apache/gobblin/service/modules/orchestration/DagManager.java [hunk 1]
```diff
@@ -1200,6 +1200,7 @@ private synchronized void addFailedDag(String dagId) {
      * @param dagId
      */
     private synchronized void cleanUpDag(String dagId) {
+      this.dags.get(dagId).setFlowEvent(null);
```
Comment (jack-moseley) [inline gobblin-service/src/main/java/org/apache/gobblin/service/modules/orchestration/DagManager.java:1203]: Maybe just add a comment here to explain why this is necessary

Comment (umustafi) [reply gobblin-service/src/main/java/org/apache/gobblin/service/modules/orchestration/DagManager.java:1204]: good point, added the comment

**Review by jack-moseley — APPROVED** [review]

**Pull request merged**